common/ts/helpers/azdo-api-utils.ts

```diff
@@ -0,0 +1,57 @@
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as request from 'request';
+import { getAuthToken } from './azdo-server-utils';
+
+export interface IPropertyBag {
+  propertyName: string;
+  propertyValue: string;
+}
+
+interface IJsonPatchBody {
+  op: string;
+  path: string;
+  value: string;
+}
+
+export function addBuildProperty(properties: IPropertyBag[]): Promise<any> {
+  return new Promise((resolve, reject) => {
+    const collectionUri = tl.getVariable('System.TeamFoundationCollectionUri') + '/';
+    const teamProjectId = tl.getVariable('System.TeamProjectId') + '/';
+    const buildId = tl.getVariable('Build.BuildId');
+
+    const patchBody: IJsonPatchBody[] = [];
+
+    properties.forEach((property: IPropertyBag) => {
+      patchBody.push({
+        op: 'add',
+        path: `/${property.propertyName}`,
+        value: `${property.propertyValue}`
+      });
+    });
+
+    tl.debug(JSON.stringify(patchBody));
+
+    const options = {
+      url:
+        collectionUri +
+        teamProjectId +
+        `_apis/build/builds/${buildId}/properties?api-version=5.0-preview.1`,
```

Comment:
Is there a non-preview API that we could use?

Reply:
No, they are documented like that, even the TFS2018 U2 is in preview as well https://docs.microsoft.com/en-us/rest/api/azure/devops/build/properties/get%20build%20properties?view=azure-devops-rest-5.0